miio/integrations/vacuum/dreame/dreamevacuum_miot.py

```diff
@@ -230,54 +409,179 @@ def status(self) -> DreameVacuumStatus:
                 prop["did"]: prop["value"] if prop["code"] == 0 else None
                 for prop in self.get_properties_for_mapping(max_properties=10)
-            }
+            },
+            self.model,
         )
 
-    def send_action(self, siid, aiid, params=None):
-        """Send action to device."""
-
-        # {"did":"<mydeviceID>","siid":18,"aiid":1,"in":[{"piid":1,"value":2}]
-        if params is None:
-            params = []
-        payload = {
-            "did": f"call-{siid}-{aiid}",
-            "siid": siid,
-            "aiid": aiid,
-            "in": params,
-        }
-        return self.send("action", payload)
+    # TODO: check the actual limit for this
+    MANUAL_ROTATION_MAX = 120
+    MANUAL_ROTATION_MIN = -MANUAL_ROTATION_MAX
+    MANUAL_DISTANCE_MAX = 300
+    MANUAL_DISTANCE_MIN = -300
 
     @command()
     def start(self) -> None:
         """Start cleaning."""
-        return self.send_action(3, 1)
+        return self.call_action("start_clean")
 
     @command()
     def stop(self) -> None:
         """Stop cleaning."""
-        return self.send_action(3, 2)
+        return self.call_action("stop_clean")
 
     @command()
     def home(self) -> None:
         """Return to home."""
-        return self.send_action(2, 1)
+        return self.call_action("home")
 
     @command()
     def identify(self) -> None:
         """Locate the device (i am here)."""
-        return self.send_action(17, 1)
+        return self.call_action("locate")
 
     @command()
     def reset_mainbrush_life(self) -> None:
         """Reset main brush life."""
-        return self.send_action(26, 1)
+        return self.call_action("reset_mainbrush_life")
 
     @command()
     def reset_filter_life(self) -> None:
         """Reset filter life."""
-        return self.send_action(27, 1)
+        return self.call_action("reset_filter_life")
 
     @command()
     def reset_sidebrush_life(self) -> None:
         """Reset side brush life."""
-        return self.send_action(28, 1)
+        return self.call_action("reset_sidebrush_life")
+
+    @command()
+    def play_sound(self) -> None:
+        """Play sound."""
+        return self.call_action("play_sound")
+
+    @command()
+    def fan_speed(self):
+        """Return fan speed."""
+        dreame_vacuum_status = self.status()
+        fanspeed = dreame_vacuum_status.cleaning_mode
+        if not fanspeed or fanspeed.value == -1:
+            _LOGGER.warning("Unknown fanspeed value received")
+            return
+        return {fanspeed.name: fanspeed.value}
+
+    @command(click.argument("speed", type=int))
+    def set_fan_speed(self, speed: int):
+        """Set fan speed.
+
+        :param int speed: Fan speed to set
+        """
+        fanspeeds_enum = self.get_fanspeeds()
+        fanspeed = None
+        if not fanspeeds_enum:
+            return
+        try:
+            fanspeed = fanspeeds_enum(speed)
+        except ValueError:
+            return None
```

Comment:
```suggestion
```
No need to check if the speed exists as trying to set the speed should error out if it errors.

Reply:
I've checked this and Dreame just return return code == 0.  So it is beeter just check that speed value exists in enum